### ISSUE TITLE
I can not to subscribe on change of all sections in Stores ->Configuration using event admin_system_config_changed_section

### ISSUE THREAD

**Issue by katesuykovskaya** [magento/magento2#5035]
## Steps to reproduce
1. M 2.0.7
2. Create extension, create etc/adminhtml/system.xml there
3. add section tab with id, that contains number, in my case "b2b" https://drive.google.com/a/interiorwebdesign.com/file/d/0BzTvZltZJMLlemNCOGVwdUM5S00/view?usp=drivesdk
4. create etc/adminhtml/events.xml and add event listener to listen to changing of this section https://drive.google.com/a/interiorwebdesign.com/file/d/0BzTvZltZJMLlRFpqOEIyc2RKRW8/view?usp=drivesdk
## Actual result
1. Error message Invalid XML in file D:/OpenServer/domains/magento207.loc/app/code/IWD/SalesRep/etc/adminhtml/events.xml:
   Element 'event', attribute 'name': [facet 'pattern'] The value 'admin_system_config_changed_section_b2b' is not accepted by the pattern '[a-zA-Z_]+'.
   Line: 3

Element 'event', attribute 'name': 'admin_system_config_changed_section_b2b' is not a valid value of the atomic type 'eventName'.
Line: 3

Element 'event', attribute 'name': Warning: No precomputed value available, the value was either invalid or something strange happend.
Line: 3
## Expected result
1. There should be ability to subcsribe on changing of each section, so I see 2 possible options:
   - extend regex to allow numbers be in name of event, or
   - disallow to set name of section, that contains numbers
     In other words, patterns, that are used to validate there values should be compatible.

**Comment by mtauqir** [comment]
I am on 2.1.4 version of magento. 
I am still seeing this validation error but on events.xml file validation. 

I am trying to observe an event from a third part module. The event has a number in its name. That number is in actually third party company's name.

Strange thing is that they are able to raise this event with numbers in its name, but XML validation prevents me to observe it. And obviously due to copyrights I am not allowed to edit their code to rename event name.

**Comment by orlangur** [comment]
This was fixed in https://github.com/magento/magento2/pull/10837 for `2.3.x`.

**Comment by okorshenko** [comment]
Hi @katesuykovskaya. Thank you for your report. 
The issue has been fixed in magento/magento2#12758 by @Mobecls in 2.2-develop branch
Related commits:
 - 937b26d3cc23ce86df6db204765b10f4ccf7ea5a
 - 85fa67170f413ea5f509c7f1e0cf870cae559bb3

The fix will be available with the upcoming patch release.

**Comment by magento-engcom-team** [comment]
Hi @katesuykovskaya. Thank you for your report. 
The issue has been fixed in magento-engcom/magento2ce#1362 by @magento-engcom-team in 2.3-develop branch
Related commit(s):
 - 4705e23479eae053947c52ee5be6969a79a82079
 - 6d355e033e513cefc390f26417eb2321756b1dce
 - 77f34cf7e3e12f567aa8fdd6918bcf9cd1ffda2f
 - 748ab4db199379ce2154ffb63b9801a8c0d13bb4
 - 90f9d95056b1cd7fe0be036ec72c236bc03738fc
 - 44ca24fdc0d3367515893cf9a8fc1bffcc070140
 - f0b7f2b1708218a8d894178fc9a4f045dfdbdedd
 - c3e1d93db404b63dc38050f9089d2b6a72094085
 - 4810755a4751e3a389e2a39dcbaf55ab72c2149c
 - 4709a222e58918876e2f5b9d28b8f0e5e120d52f
 - a2b9fab4b10421c327b1f323bf222750ac371fb9
 - ac7df33cecf744cf9899067dc62453bc7a15ca50
 - 245238dcbb72b10fad84596399b12b9814295641
 - 2c153c3c428308c1cbc1969bd2eb271913d6551d
 - 48a72fa0d214418a79a5d77d25ead1eb75cbc1c1
 - 625f685c183e0c0a7993ca38c025a079e0e29b47
 - afa4516615808a427be8a160e9cd6ae80c321b27

The fix will be available with the upcoming 2.3.0 release.
